Source/Dafny/Compilers/Compiler-go.cs

```diff
@@ -3519,6 +3520,14 @@ private class SpecialNativeType : UserDefinedType {
       // Dafny compiles to the old Go package system, whereas Go has moved on to a module
       // system. Until Dafny's Go compiler catches up, the GO111MODULE variable has to be set.
       psi.EnvironmentVariables["GO111MODULE"] = "auto";
+      if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+        // On Windows, Path.GetTempPath() returns "c:\Windows" which, being not writable, crashes Go.
+        // Hence we set up a local temporary directory depending on the user.
+        psi.EnvironmentVariables["GOTMPDIR"] =
+          Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData) + "\\Temp";
```

Comment:
Should we clean that up?

Reply:
What do you mean? These four lines of code literally fix all the GO tests on Windows.
I took action on this by refactoring `Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)` into a single variable, but I'm not sure if it's what you meant.

Reply:
I was wondering whether this temporary data is still there after finishing the compilation.

Reply:
Oh, I see, it's a general question about the temporary directory, not specific about Windows. Here we only provide a valid path for temporary files, but Unix and MacOX use their own paths. Here for Windows I use a standard temp directory that is weirdly not used by `Path.GetTemporaryPath`.
I don't think we should clean up the temporary directory ourselves. It might be used at any time by other processes.